addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8777,20 +8780,77 @@ registry.many2one = SnippetOptionWidget.extend({
  * Allows to display a warning message on outdated snippets.
  */
 registry.VersionControl = SnippetOptionWidget.extend({
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Replaces an outdated snippet by its new version.
+     */
+    async replaceSnippet() {
+        // Getting the new block version.
+        const snippet = this.$target[0].dataset.snippet;
+        const snippetTagName = this.$target[0].tagName.toLowerCase();
+        const customizePanelEl = this.$el[0].closest(".o_we_customize_panel");
+        const newBlockEl = customizePanelEl.parentElement
+            .querySelector(`${snippetTagName}[data-snippet=${snippet}]`).cloneNode(true);
+        // Replacing the block.
+        this.options.wysiwyg.odooEditor.historyPauseSteps();
+        this.$target[0].classList.add("d-none"); // Hiding the block to replace it smoothly.
+        this.$target[0].insertAdjacentElement("beforebegin", newBlockEl);
+        // Initializing the new block as if it was dropped: the mutex needs to
+        // be free for that so we wait for it first.
+        this.options.wysiwyg.waitForEmptyMutexAction().then(async () => {
+            await this.options.wysiwyg.snippetsMenu.callPostSnippetDrop($(newBlockEl));
```

Comment:
Probably will be reviewed once we go through OWL. For now, this seems good, especially since we are talking about replacing old snippets which cannot be currently being edited.